### DISCUSSION TITLE
[feature] Add a shimmer state to Clarity cards

### DISCUSSION THREAD

**Discussion by majid-frz** [vmware/clarity#5013] [Ideas]
## Summary

It would be nice if Clarity cards had a shimmer effect state that would be controlled with a boolean and in pages that we are waiting for server's data those cards would had their place but with a shimmer effect so that the user would know they are loading.

## Use case

For example on a products page that use Clarity cards, those cards would already be in the page before receiving the data from server but with a nice shimmer effect.

## Examples

There is a popular NPM package for shimmer effect called [ngx-shimmer-loading](https://www.npmjs.com/package/ngx-shimmer-loading). Here's their [Demo page](https://ahmedbeheiry.github.io/ngx-shimmer-loading/).

## Workarounds

Using third-party NPM packages or using Clarity spinners.

**Comment by gnomeontherun** [comment]
I'm going to convert this to a GitHub discussion. There isn't much that Clarity needs to do here if you want to use a shimmer feature on your end, so I don't really see how this makes sense as a fully supported feature in the card. What would the layout be? How should it appear? That depends on what you put in the card, and that is in the hands of the apps.
